File: deporvida_backend/src/main/java/com/deporvida/service/CarritoService.java
package com.deporvida.service;

import com.deporvida.dto.request.ItemCarritoRequest;
import com.deporvida.dto.response.CarritoResponse;

public interface CarritoService {

    CarritoResponse getCarrito(Long usuarioId);

    CarritoResponse addItem(Long usuarioId, ItemCarritoRequest request);

    CarritoResponse updateItem(Long usuarioId, Long itemId, Integer cantidad);

    CarritoResponse removeItem(Long usuarioId, Long itemId);

    void clearCarrito(Long usuarioId);

    int getItemCount(Long usuarioId);
}

File: src/services/carritoService.ts
import api from './api';
import type { LibroResponse } from './libroService';

export interface ItemCarritoRequest {
  libroId: number;
  cantidad: number;
}

export interface ItemCarritoResponse {
  id: number;
  libro: LibroResponse;
  cantidad: number;
  precioUnitario: number;
}

export interface CarritoResponse {
  id: number;
  items: ItemCarritoResponse[];
  cantidadItems: number;
  subtotal: number;
}

export const carritoService = {
  getCarrito: () =>
    api.get<CarritoResponse>('/carrito'),

  addItem: (data: ItemCarritoRequest) =>
    api.post<CarritoResponse>('/carrito/items', data),

  updateItem: (itemId: number, cantidad: number) =>
    api.put<CarritoResponse>(`/carrito/items/${itemId}`, null, { params: { cantidad } }),

  removeItem: (itemId: number) =>
    api.delete<CarritoResponse>(`/carrito/items/${itemId}`),

  clearCarrito: () =>
    api.delete('/carrito'),
};
